test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+        await uploadFilesAndCheck(httpClient, filesToPut);
+        const href = filesToPut[0].resourceHref;
+        const basePath = href.substring(0, href.lastIndexOf("/"));
+        try {
+          await doPutFiles(httpClient, basePath, filesToPutInvalid);
+          assert.fail('An error is expected !')
+        } catch (err) {
+          assert.equal(err.code, 'streaming.body.read.failed');
+          const partialBodyStr = err.partialBody.toString();
+          assert.equal(partialBodyStr.includes('"status": 400'), true, 'expected status code 400 missing in partial body');
+          assert.equal(partialBodyStr.includes('"code": "upload.and.copy.mix"'), true, 'expected error code missing in partial body');
+        }
+      });
+    });
+
+    describe("customRouteForGet", function () {
+      it("/resource/:key/attachments/:attachmentKey should work", async () => {
+        // this function implicitly tests functions passed to
+        // customRouteForUpload, customRouteForUploadCopy, customRouteForDelete and customRouteForGet
+
+        const body = {
+          type: "person",
+          name: "test user",
+          status: "active",
+        };
+        const resourceKey = uuid.v4();
+        const resourceHref = type + "/" + resourceKey;
+        const attachmentKey = uuid.v4();
+        const localFilename = "test/images/orange-boy-icon.png";
+        const attachmentUrl = `${type}/${resourceKey}/attachments/${attachmentKey}`;
+        const attachmentDownloadUrl =
+          type + "/" + resourceKey + "/attachments/profile.png";
+
+        const responsePut = await httpClient.put({ path: resourceHref, body });
+        assert.equal(responsePut.status, 201);
+
+        // Add attachment
+        debug("PUTting the profile image as attachment");
+        const filesToPut = [
+          {
+            remotefileName: "profile.png",
+            localFilename,
+            attachmentKey,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut);
+
+        const responseGetAtt1 = await httpClient.get({ path: 
+          `${type}/${resourceKey}/attachments/${attachmentKey}`
+        });
+        assert.equal(responseGetAtt1.status, 200);
+        assert.equal(
+          responseGetAtt1.body.description,
+          `this is MY file with key ${attachmentKey}`
+        );
+
+        // Copy resource and attachment
+        const resourceCopyKey = uuid.v4();
+        const resourceCopyHref = type + "/" + resourceCopyKey;
+        const copyResponse = await httpClient.put({ path: resourceCopyHref, body });
+        assert.equal(copyResponse.status, 201);
+
+        const attachmentCopyKey = uuid.v4();
+
+        const FileToCopy =
+        [{
+          remotefileName: "profile1.png",
+          urlToCopy: attachmentDownloadUrl,
+          attachmentKey: attachmentCopyKey,
+          resourceHref: resourceCopyHref,
+        }];
+        await copyFilesAndCheck(httpClient, FileToCopy);
```

Comment:
Redundant check?

Reply:
Similar code has indeed been checked in earlier tests, but I prefer to keep the check so that in case of an error it is clear where in the test it went wrong.